File: educational-core/src/com/jetbrains/edu/learning/checkio/api/exceptions/HttpException.java
package com.jetbrains.edu.learning.checkio.api.exceptions;

import com.jetbrains.edu.learning.checkio.call.CheckiOCall;
import org.jetbrains.annotations.NotNull;
import retrofit2.Response;

/**
 * This exception is thrown when unexpected non-2xx HTTP response is received
 * Similar to {@link retrofit2.HttpException}, but checked
 *
 * @see CheckiOCall#execute()
 * */
public class HttpException extends ApiException {
  public HttpException(@NotNull Response<?> response) {
    super("HTTP " + response.code() + " " + response.message());
  }
}

File: educational-core/src/com/jetbrains/edu/learning/checkio/api/exceptions/HttpException.kt
package com.jetbrains.edu.learning.checkio.api.exceptions

import com.jetbrains.edu.learning.messages.EduCoreBundle.message
import retrofit2.Response

/**
 * This exception is thrown when unexpected non-2xx HTTP response is received
 * Similar to [retrofit2.HttpException], but checked
 */
class HttpException(val response: Response<*>) :
  ApiException(message("exception.message.http.info", response.code(), response.message()))
